src/main.rs

```diff
@@ -161,11 +161,24 @@ fn run() -> Result<(), failure::Error> {
         println!("Enter email: ");
         let mut email: String = read!("{}\n");
         email.truncate(email.trim_end().len());
-        println!("Enter api key: ");
-        let mut api_key: String = read!("{}\n");
-        api_key.truncate(api_key.trim_end().len());
 
-        commands::global_config(email, api_key)?;
+        println!("If using API key, enter API key (otherwise press Enter): ");
+        let mut api_key_str: String = read!("{}\n");
+        api_key_str.truncate(api_key_str.trim_end().len());
+        let mut api_key = None;
+        if !api_key_str.is_empty() {
+            api_key = Some(api_key_str);
+        }
+
+        println!("If using API token, enter API token (otherwise press Enter): ");
+        let mut api_token_str: String = read!("{}\n");
+        api_token_str.truncate(api_token_str.trim_end().len());
+        let mut api_token = None;
+        if !api_token_str.is_empty() {
+            api_token = Some(api_token_str);
+        }
+
+        commands::global_config(email, api_key, api_token)?;
```

Comment:
What if the user specifies both an API key and an API token? They will be send both in headers (unless you implement my suggestion  https://github.com/cloudflare/wrangler/pull/471/files#r318516334).
Maye we should restrict that?

Reply:
you should prefer api token. that is what we do in terraform